### PR TITLE
Feat: Support unset index by hash

### PR DESCRIPTION
**Pull request by guysl10**

### CODE REVIEW

File: intezer_sdk/consts.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ class IndexStatusCode(enum.Enum):
     IN_PROGRESS = 'in_progress'
     FINISH = 'finished'
     FINISHED = 'finished'
+    UNSET = 'unset'
```
Comment (davidt99) [inline intezer_sdk/consts.py:66]: This is the status of the operation, not the status of the private index

File: tests/unit/test_index.py [hunk 2]
```diff
@@ -241,3 +241,24 @@ def test_parallel_index_by_sha256_succeeded_status_changed_to_finish(self):
...
         self.assertEqual(second_index.status, consts.IndexStatusCode.FINISHED)
+
+    def test_unset_index_type(self):
+        # Arrange
+        sha256 = 'a'
+        index = Index(sha256=sha256, index_as=consts.IndexType.TRUSTED)
+        with responses.RequestsMock() as mock:
+            mock.add('DELETE',
+                     url=self.full_url + f'/files/{sha256}/index',
+                     status=200)
```
Comment (davidt99) [inline tests/unit/test_index.py:252]: Use HTTPStatus

File: tests/unit/test_index.py [hunk 3]
```diff
@@ -241,3 +241,24 @@ def test_parallel_index_by_sha256_succeeded_status_changed_to_finish(self):
...
+            index.unset_indexing(wait=True)
+
+    def test_unset_index_type_raises_error(self):
+        # Arrange
+        sha256 = 'a'
+        index = Index(sha256=sha256, index_as=consts.IndexType.TRUSTED)
+        with responses.RequestsMock() as mock:
+            mock.add('DELETE',
+                     url=self.full_url + f'/files/{sha256}/index',
+                     status=404)
```
Comment (davidt99) [inline tests/unit/test_index.py:262]: Same

File: intezer_sdk/_api.py [hunk 4]
```diff
@@ -347,6 +347,12 @@ def index_by_sha256(self, sha256: str, index_as: IndexType, family_name: str = N
 
         return self._get_index_id_from_response(response)
 
+    def unset_index_by_sha256(self, sha256: str):
+        response = self.api.request_with_refresh_expired_access_token('DELETE', f'/files/{sha256}/index')
```
Comment (davidt99) [inline intezer_sdk/_api.py:351]: Document this function, and if you can, document all of the function in this module

**Review by davidt99 — COMMENTED** [review]
Missing changes and version bump

File: intezer_sdk/index.py [hunk 5]
```diff
@@ -16,6 +16,15 @@ def __init__(self,
                  sha256: str = None,
                  api: IntezerApiClient = None,
                  family_name: str = None):
+        """
+        Index a file or a sha256.
+
+        :param index_as: The type of the index.
```
Comment (davidt99) [inline intezer_sdk/index.py:22]: Mention the two values

**Review by davidt99 — APPROVED** [review]

**Review by ofirit — APPROVED** [review]

**Pull request merged**